### PR TITLE
UserUpdate endpoint

### PR DESCRIPTION
**Pull request by Schparky**

### CODE REVIEW

File: application/actions/users.go [hunk 1]
```diff
@@ -31,6 +32,68 @@ func usersMe(c buffalo.Context) error {
 	return c.Render(http.StatusOK, r.JSON(output))
 }
 
+// swagger:operation PUT /users/me Users UsersMe
```
Comment (Baggerone) [inline application/actions/users.go:35]: Should the last word include `Update` in it?

File: application/actions/users_test.go [hunk 2]
```diff
@@ -15,10 +18,57 @@ func (as *ActionSuite) Test_convertUserToPrivateAPIType() {
...
+		fmt.Sprintf(`"photo_id":"%s"`, photoID),
+		fmt.Sprintf(`"nickname":"%s"`, nickname),
+	}
+	for _, w := range wantContains {
+		as.Contains(body, w)
+	}
+
+	// test for removing photo
+	reqBody = api.UsersInput{
+		// remote the photo by leaving it as nil
```
Comment (Baggerone) [inline application/actions/users_test.go:61]:
```suggestion
		// remove the photo by leaving it as nil
```

**Review by Baggerone — APPROVED** [review]

File: application/models/user.go [hunk 3]
```diff
@@ -512,7 +513,21 @@ func (u *User) GetPhotoURL(tx *pop.Connection) (*string, error) {
 // Save wraps tx.Save() call to check for errors and operate on attached object
 func (u *User) Save(tx *pop.Connection) error {
 	u.Nickname = domain.RemoveUnwantedChars(u.Nickname, "-_ .,'&@")
-	return save(tx, u)
+	if err := save(tx, u); err != nil {
+		appError := api.AppError{Err: err}
+		if strings.Contains(err.Error(), "Nickname must have a visible character") {
+			appError.Key = "UpdateUser.InvisibleNickname"
```
Comment (Baggerone) [inline application/models/user.go:519]: Should this be a const in the api package?

Comment (Schparky) [reply application/models/user.go:519]: Yep, thanks

File: application/actions/users.go (hunk 1 above)
Comment (Schparky) [reply application/actions/users.go:35]: Yep

**Pull request merged**